### PR TITLE
Cw 830 coin control getting cleared

### PR DESCRIPTION
**Pull request by Serhii-Borodenko**
Issue Number (if Applicable): Fixes #

# Description

Please include a summary of the changes and which issue is fixed / feature is added. 

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods

### CODE REVIEW

File: cw_core/lib/unspent_comparable_mixin.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  bool isSending = true;
+
+  String note = '';
+
+  void updateAdjustableFieldsFrom(UnspentComparable other) {
+    isFrozen = other.isFrozen;
+    isSending = other.isSending;
+    note = other.note;
+  }
```
Comment (OmarHatem28) [inline cw_core/lib/unspent_comparable_mixin.dart:23]: like this, would it modify the variables here or in the main object?
I don't think this approach is ideal

File: lib/src/screens/unspent_coins/unspent_coins_list_page.dart [hunk 2]
```diff
@@ -1,30 +1,68 @@
...
+  }
+
   final UnspentCoinsListViewModel unspentCoinsListViewModel;
 
+  Future<void> handleOnPopInvoked(BuildContext context) async {
+    final hasChanged = unspentCoinsListViewModel.hasAdjustableFieldChanged;
+    if (unspentCoinsListViewModel.items.isEmpty || !hasChanged) {
+      Navigator.of(context).pop();
+    } else {
+      unspentCoinsListViewModel.isDisposing = true;
```
Comment (OmarHatem28) [inline lib/src/screens/unspent_coins/unspent_coins_list_page.dart:49]: would this trigger the reaction even tho it's not an @action? or @action is only needed to trigger the Observer widget builder?

File: lib/src/screens/unspent_coins/unspent_coins_list_page.dart [hunk 3]
```diff
@@ -1,30 +1,68 @@
...
+  Widget body(BuildContext context) =>
+      UnspentCoinsListForm(unspentCoinsListViewModel, handleOnPopInvoked);
 }
 
 class UnspentCoinsListForm extends StatefulWidget {
-  UnspentCoinsListForm(this.unspentCoinsListViewModel);
+  UnspentCoinsListForm(this.unspentCoinsListViewModel, this.handleOnPopInvoked);
 
   final UnspentCoinsListViewModel unspentCoinsListViewModel;
+  final Future<void> Function(BuildContext context) handleOnPopInvoked;
```
Comment (OmarHatem28) [inline lib/src/screens/unspent_coins/unspent_coins_list_page.dart:65]: **minor**
can be ignored since you already use `widget.handleOnPopInvoked`

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]

File: lib/src/screens/unspent_coins/unspent_coins_list_page.dart (hunk 2 above)
Comment (Serhii-Borodenko) [reply lib/src/screens/unspent_coins/unspent_coins_list_page.dart:49]: yes, it is for triggering a reaction. It is working as is, but I’ve added an action to make it clearer

File: cw_core/lib/unspent_comparable_mixin.dart (hunk 1 above)
Comment (Serhii-Borodenko) [reply cw_core/lib/unspent_comparable_mixin.dart:23]: reverted

File: lib/src/screens/unspent_coins/unspent_coins_list_page.dart (hunk 3 above)
Comment (Serhii-Borodenko) [reply lib/src/screens/unspent_coins/unspent_coins_list_page.dart:65]: UnspentCoinsListForm is a separate widget, so it should be there

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**